miniapps/dpg/pmaxwell.cpp

```diff
@@ -0,0 +1,1444 @@
+// | G | -iωϵ (β E,G) |   (H,∇ × G)      |              | < n × Ĥ, G > |  (J,G)  |
+// where (F,G) ∈  H(curl,Ω) × H(curl,Ω)
+
+// For more information see https://doi.org/10.1016/j.camwa.2021.01.017
+
+#include "mfem.hpp"
+#include "util/pcomplexweakform.hpp"
+#include "util/pml.hpp"
+#include "../common/mfem-common.hpp"
+#include <fstream>
+#include <iostream>
+
+using namespace std;
+using namespace mfem;
+using namespace mfem::common;
+
+void E_exact_r(const Vector &x, Vector & E_r);
+void E_exact_i(const Vector &x, Vector & E_i);
+
+void H_exact_r(const Vector &x, Vector & H_r);
+void H_exact_i(const Vector &x, Vector & H_i);
+
+
+void  rhs_func_r(const Vector &x, Vector & J_r);
+void  rhs_func_i(const Vector &x, Vector & J_i);
+
+void curlE_exact_r(const Vector &x, Vector &curlE_r);
+void curlE_exact_i(const Vector &x, Vector &curlE_i);
+void curlH_exact_r(const Vector &x,Vector &curlH_r);
+void curlH_exact_i(const Vector &x,Vector &curlH_i);
+
+void curlcurlE_exact_r(const Vector &x, Vector & curlcurlE_r);
+void curlcurlE_exact_i(const Vector &x, Vector & curlcurlE_i);
+
+void hatE_exact_r(const Vector & X, Vector & hatE_r);
+void hatE_exact_i(const Vector & X, Vector & hatE_i);
+
+void hatH_exact_r(const Vector & X, Vector & hatH_r);
+void hatH_exact_i(const Vector & X, Vector & hatH_i);
+
+double hatH_exact_scalar_r(const Vector & X);
+double hatH_exact_scalar_i(const Vector & X);
+
+void maxwell_solution(const Vector & X,
+                      std::vector<complex<double>> &E);
+
+void maxwell_solution_curl(const Vector & X,
+                           std::vector<complex<double>> &curlE);
+
+void maxwell_solution_curlcurl(const Vector & X,
+                               std::vector<complex<double>> &curlcurlE);
+
+void source_function(const Vector &x, Vector & f);
+
+int dim;
+int dimc;
+double omega;
+double mu = 1.0;
+double epsilon = 1.0;
+
+enum prob_type
+{
+   plane_wave,
+   fichera_oven,
+   pml_general,
+   pml_plane_wave_scatter,
+   pml_pointsource
+};
+
+static const char *enum_str[] =
+{
+   "plane_wave",
+   "fichera_oven",
+   "pml_general",
+   "pml_plane_wave_scatter",
+   "pml_pointsource"
+};
+
+prob_type prob;
+
+int main(int argc, char *argv[])
+{
+   Mpi::Init();
+   int myid = Mpi::WorldRank();
+   Hypre::Init();
+
+   const char *mesh_file = "../../data/inline-quad.mesh";
+   int order = 1;
+   int delta_order = 1;
+   double rnum=1.0;
+   double theta = 0.0;
+   bool static_cond = false;
+   int iprob = 0;
+   int sr = 0;
+   int pr = 1;
+   bool exact_known = false;
+   bool with_pml = false;
+   bool visualization = true;
+   bool paraview = false;
+
+   OptionsParser args(argc, argv);
+   args.AddOption(&mesh_file, "-m", "--mesh",
+                  "Mesh file to use.");
+   args.AddOption(&order, "-o", "--order",
+                  "Finite element order (polynomial degree)");
+   args.AddOption(&rnum, "-rnum", "--number_of_wavelengths",
```

Comment:
Some of these options use `_` for spaces and some use `-`. We should probably be consistent. I think `-`, i.e. dashes, are used in most examples and miniapps.